scripts/build/gen_isr_tables.py

```diff
@@ -131,12 +131,65 @@ def parse_args():
 #endif
 """
 
+sh_isr_function = "void __attribute__((section(\".shared_isr\")))shared_isr_"
+sh_isr_args = "const size_t shared_isr_"
+sh_isr_args_section = " __attribute__((section(\".shared_isr_args\")))"
+
 def get_symbol_from_addr(syms, addr):
     for key, value in syms.items():
         if addr == value:
             return key
     return None
 
+def write_shared_direct_isr_prototypes(fp, nv, shi):
+    for i in range(nv):
+        if len(shi[i]) > 0:
+            fp.write(sh_isr_function + "direct_" + str(i) + "(void);\n")
+
+def write_shared_direct_isr(fp, nv, shi):
+    for i in range(nv):
+        if len(shi[i]) > 0:
+            fp.write(sh_isr_function + "direct_" + str(i) +"(void){\n")
+            for isr in shi[i]:
+                func = str(hex(isr))
+                fp.write("\t((DIRECT_ISR)" + func + ")();\n")
+            fp.write("}\n")
+
+def write_shared_isr_prototypes(fp, nv, shi):
+    for i in range(nv):
+        if len(shi[i]) > 0:
+            fp.write(sh_isr_function + str(i) + "(void *args);\n")
+
+def write_shared_isr_args_forward_declare(fp, nv, shi):
+    for i in range(nv):
+        if len(shi[i]) > 0:
+            fp.write(sh_isr_args + str(i) + "_args[" +
+                str(len(shi[i]))  +"]" + sh_isr_args_section + ";\n")
+
+def write_shared_isr(fp, nv, shi):
+    for i in range(nv):
+        if len(shi[i]) > 0:
+            fp.write(sh_isr_function + str(i) +"(void *args){\n")
+            idx = 0
+            for isr in shi[i]:
+                func = str(hex(isr[1]))
+                fp.write("\t((ISR)" + func + ")((void *)((size_t *)args)[" +
+                    str(idx) + "]);\n")
+                idx += 1
+            fp.write("}\n")
+
+def write_shared_isr_args(fp, nv, shi):
+    for i in range(nv):
+        if len(shi[i]) > 0:
+            fp.write("const size_t shared_isr_" + str(i) + "_args[" +
```

Comment:
Why args are now `size_t`?

Reply:
I thought size_t would be more portable when this feature is enabled on other architectures.

Reply:
Please stick to `(const void *)`